src/KeyVault/KeyVault/SecurityDomain/Cmdlets/RestoreSecurityDomain.cs

```diff
@@ -1,49 +1,159 @@
+        protected const string DoRestoreBlob = "DoRestoreBlob";
+        protected const string ByRestoredBlob = "ByRestoredBlob";
+        protected const string GenerateExchangeKey = "GenerateExchangeKey";
+        protected const string ByName = "ByName";
+        protected const string ByInputObject = "ByInputObject";
+        // protected const string ByResourceId = "ByResourceID";
+
+        [Parameter(HelpMessage = "Name of the managed HSM.", Mandatory = true, ParameterSetName = ByName)]
+        [Parameter(HelpMessage = "Name of the managed HSM.", Mandatory = true, ParameterSetName = ByRestoredBlob)]
+        [Parameter(HelpMessage = "Name of the managed HSM.", Mandatory = true, ParameterSetName = GenerateExchangeKey)]
+        [Alias("HsmName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(HelpMessage = "Object representing a managed HSM.", Mandatory = true, ParameterSetName = ByInputObject, ValueFromPipeline = true)]
+        [ValidateNotNull]
+        public PSKeyVaultIdentityItem InputObject { get; set; }
+
+        [Parameter(HelpMessage = "Information about the keys that are used to decrypt the security domain data. See examples for how it is constructed.", Mandatory = true, ParameterSetName = ByName)]
+        [Parameter(HelpMessage = "Information about the keys that are used to decrypt the security domain data. See examples for how it is constructed.", Mandatory = true, ParameterSetName = ByInputObject)]
+        [Parameter(HelpMessage = "Information about the keys that are used to decrypt the security domain data. See examples for how it is constructed.", Mandatory = true, ParameterSetName = DoRestoreBlob)]
         [ValidateNotNullOrEmpty]
         public KeyPath[] Keys { get; set; }
 
-        [Parameter(HelpMessage = "Specify the path to the encrypted security domain data.", Mandatory = true)]
+        [Parameter(HelpMessage = "Specify the path to the encrypted security domain data.", Mandatory = true, ParameterSetName = ByName)]
+        [Parameter(HelpMessage = "Specify the path to the encrypted security domain data.", Mandatory = true, ParameterSetName = ByInputObject)]
+        [Parameter(HelpMessage = "Specify the path to the prepared security domain data.", Mandatory = true, ParameterSetName = ByRestoredBlob)]
+        [Parameter(HelpMessage = "Specify the path to the encrypted security domain data.", Mandatory = true, ParameterSetName = DoRestoreBlob)]
         [Alias("Path")]
         [ValidateNotNullOrEmpty]
         public string SecurityDomainPath { get; set; }
 
+        [Parameter(HelpMessage = "Local file path to store the security domain encrypted with the exchange key.", Mandatory = true, ParameterSetName = DoRestoreBlob)]
+        [ValidateNotNullOrEmpty]
+        public string SecurityDomainRestoredBlob { get; set; }
```

Comment:
```suggestion
        public string RestoredBlobPath { get; set; }
```

Reply:
Better not change to align with `SecurityDomainPath`. Not prepared sd data - `SecurityDomainPath`, prepared sd dara - `SecurityDomainRestoredBlob`. They are all files containing sd data.

Reply:
SecurityDomainRestoredBlob  means the path of RestoredBlob. It's reasonable to add Path in its naming to avoid confusion.  SecurityDomainRestoredBlobPath is fine for me. But I think SecurityDomain can be omitted in the context of `Import-AzKeyVaultSecurityDomain`. Removing SecurityDomain from SecurityDomainPath is not applicable because Path seems lacking semantic meaning.

Reply:
`Blob` itself has the meaning of `Path`